packages/@react-spectrum/provider/docs/Provider.mdx

```diff
@@ -146,6 +146,42 @@ function Register() {
 }
 ```
 
+## useProvider
+<TypeLink links={docs.links} type={docs.exports.useProvider} /> can be used to access the various properties applied by the nearest parent Provider. This can be useful for adapting a component's various features, such as its icon, to the current color scheme or
+scale.
+
+The example below uses the `colorScheme` sourced from `useProvider` to determine the top button's icon when the docs theme is flipped. Note how the icon of the ActionButton within the nested Provider doesn't change when you toggle the theme switcher since the `colorScheme` for its
+parent Provider is fixed.
+
+```tsx example
+import Light from '@spectrum-icons/workflow/Light';
+import Moon from '@spectrum-icons/workflow/Moon';
+import {useProvider} from '@react-spectrum/provider';
+
+function ColorSchemeButton() {
+  let {colorScheme} = useProvider();
+  let label = colorScheme === 'dark' ? 'Moon icon button for dark theme' : 'Light icon button for light theme';
+  return (
+    <ActionButton aria-label={label} margin="size-100">
+      {colorScheme === 'dark' ? <Moon /> : <Light />}
+    </ActionButton>
+  );
+}
+
+function Example() {
+  return (
+    <Flex direction="column" gap="size-100" alignItems="start">
+      <ColorSchemeButton />
+      <Provider colorScheme="light">
+        <ColorSchemeButton />
+      </Provider>
+    </Flex>
+  );
+}
+
+<Example />
+```
```

Comment:
Example kinda sucks, will try to think of a different one that is relatively simple still

Reply:
I've pushed two slightly different examples, lemme know which one you like better.